app/validators/incomplete_undergraduate_course_details_validator.rb

```diff
@@ -0,0 +1,24 @@
+class IncompleteUndergraduateCourseDetailsValidator < ActiveModel::EachValidator
+  include ActionView::Helpers::UrlHelper
```

Comment:
Do you need this? You should be able to call `Rails.application.routes.url_helpers.candidate_interface_other_qualification_type_path` without it. Or if you want to include it, just call `candidate_interface_other_qualification_type_path` directly.